addons/survey/models/survey_user_input.py

```diff
@@ -782,3 +783,41 @@ def _get_answer_score_values(self, vals, compute_speed_score=True):
             'answer_is_correct': answer_is_correct,
             'answer_score': answer_score
         }
+
+    @api.model
+    def _get_answer_value_field(self):
+        if self.answer_type == 'char_box':
+            return 'value_char_box'
+        elif self.answer_type == 'text_box':
+            return 'value_text_box'
+        elif self.answer_type == 'numerical_box':
+            return 'value_numerical_box'
+        elif self.answer_type == 'date':
+            return 'value_date'
+        elif self.answer_type == 'datetime':
+            return 'value_datetime'
+        elif self.answer_type == 'suggestion':
+            return 'suggested_answer_id.value'
+
+    @api.model
+    def _get_matching_answer_domain(self, operator, value):
+        self.ensure_one()
+        value_field = self._get_answer_value_field()
+        if self.answer_type in ('char_box', 'text_box', 'numerical_box', 'date', 'datetime'):
+            return [expression.AND([[('question_id', '=', self.question_id.id)], [(value_field, operator, value)]])]
```

Comment:
We could do "elif suggestion" and redirect to the "_get_matching_answer_domain" on the related answer_id / row_id ?

Even if not used, makes the API more robust.